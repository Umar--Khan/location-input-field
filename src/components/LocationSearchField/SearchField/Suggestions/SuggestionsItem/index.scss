.suggestions-list-item {
  font-weight: 500;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  padding: 10px;
  border-bottom: 1px solid $color-grey;

  &:hover {
    background-color: $color-light-blue;
  }

  &__content {
    padding-left: 8px;
  }

  &--font-emphasized {
    font-size: 14px;
  }

  &--font-caption {
    font-size: 12px;
  }
}
