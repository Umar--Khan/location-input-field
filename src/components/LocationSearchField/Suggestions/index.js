const noResultsMessage = 'No results found';

const Suggestions = ({ suggestionsResults, isActive }) => {
  const renderSuggestionsResults = () =>
    suggestionsResults.map(suggestion => {
      if (suggestion.name === noResultsMessage) {
        return <li key={suggestion.index}>{noResultsMessage}</li>;
      }

      return (
        <li className='suggestions-list-item' key={suggestion.placeKey}>
          <div className='suggestions-list-item--font-emphasized'>
            {suggestion.name}
          </div>
          <div className='suggestions-list-item--font-caption'>
            {suggestion.country}
          </div>
        </li>
      );
    });

  return (
    <div className={`suggestions-list-container${isActive ? ' active' : ''}`}>
      <ul className='suggestions-list'>{renderSuggestionsResults()}</ul>
    </div>
  );
};

export default Suggestions;
