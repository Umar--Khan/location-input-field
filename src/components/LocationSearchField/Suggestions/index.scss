.suggestions-list-container {
  display: none;
  position: absolute;
  right: 0;
  left: 0;
  background-color: white;
  top: $space-090;
}

.suggestions-list {
  list-style: none;
  padding: 0;
  margin: 0;
}

.active {
  display: block;
}
