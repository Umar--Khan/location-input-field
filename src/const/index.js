export const NO_RESULTS_FOUND = 'No results found';

export const PLACE_TYPES_LABELS = {
  A: 'Airport',
  D: 'District',
  C: 'City',
  S: 'Station',
};

export const PLACE_TYPES_COLORS = {
  A: 'maroon',
  D: 'green',
  C: 'blue',
  S: 'dark-grey',
};
