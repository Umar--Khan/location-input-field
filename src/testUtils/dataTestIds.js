export const page = {
  pageTitle: 'page-title',
  locationSearchFieldTitle: 'location-search-field-title',
};

export const tags = {
  tagLabel: 'tags-label',
};

export const suggestions = {
  suggestionItem: 'suggestion-item',
  suggestionItemCountry: 'suggestion-item-country',
  suggestionItemName: 'suggestion-item-name',
  container: 'suggestions-container',
};

export const searchField = {
  inputField: 'search-input-field',
  inputFieldLabel: 'search-input-field-label',
};

export const app = {
  appContainer: 'app-container',
};
