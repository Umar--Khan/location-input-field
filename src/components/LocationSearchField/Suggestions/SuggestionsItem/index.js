import { PLACE_TYPES_LABELS, PLACE_TYPES_COLORS } from 'const';
import Tag from 'shared/Tag';

const renderTag = placeType => {
  const label = PLACE_TYPES_LABELS[placeType];
  const color = PLACE_TYPES_COLORS[placeType];

  return <Tag color={color} label={label} />;
};

const SuggestionsItem = ({ suggestion }) => (
  <li
    className='suggestions-list-item'
    key={suggestion.placeKey}
    id={`suggestions-list-item-${suggestion.index}`}
    role='option'
    aria-selected='false'
  >
    {renderTag(suggestion.placeType)}
    <div className='suggestions-list-item__content'>
      <p className='suggestions-list-item--font-emphasized'>
        {suggestion.name}
      </p>
      <p className='suggestions-list-item--font-caption'>
        {suggestion.country}
      </p>
    </div>
  </li>
);

export default SuggestionsItem;
