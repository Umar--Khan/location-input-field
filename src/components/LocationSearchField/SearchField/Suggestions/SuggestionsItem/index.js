import { PLACE_TYPES_LABELS, PLACE_TYPES_COLORS } from 'const';
import Tag from 'shared/Tag';
import { suggestions } from 'testUtils/dataTestIds';

const renderTag = placeType => {
  const label = PLACE_TYPES_LABELS[placeType];
  const color = PLACE_TYPES_COLORS[placeType];

  return <Tag color={color} label={label} />;
};

const SuggestionsItem = ({ suggestion }) => (
  <li
    className='suggestions-list-item'
    key={suggestion.placeKey}
    id={`suggestions-list-item-${suggestion.index}`}
    role='option'
    aria-selected='false'
    data-testid={suggestions.suggestionItem}
  >
    {renderTag(suggestion.placeType)}
    <div className='suggestions-list-item__content'>
      <p
        className='suggestions-list-item--font-emphasized'
        data-testid={suggestions.suggestionItemName}
      >
        {suggestion.name}
      </p>
      <p
        className='suggestions-list-item--font-caption'
        data-testid={suggestions.suggestionItemCountry}
      >
        {suggestion.country}
      </p>
    </div>
  </li>
);

export default SuggestionsItem;
