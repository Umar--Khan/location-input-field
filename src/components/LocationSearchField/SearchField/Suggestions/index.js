import SuggestionsItem from 'components/LocationSearchField/SearchField/Suggestions/SuggestionsItem';
import { NO_RESULTS_FOUND } from 'const';
import { suggestions } from 'testUtils/dataTestIds';

const Suggestions = ({ suggestionsResults, isActive }) => {
  const renderSuggestionsResults = () =>
    suggestionsResults.map(suggestion => {
      if (suggestion.name === NO_RESULTS_FOUND) {
        return (
          <li
            className='suggestions-list-item'
            id={`suggestions-list-item${suggestion.index}`}
            key={suggestion.index}
            aria-describedby='pick-up-locations-suggestions-no-results'
            data-testid={suggestions.suggestionItem}
          >
            {NO_RESULTS_FOUND}
          </li>
        );
      }

      return (
        <SuggestionsItem suggestion={suggestion} key={suggestion.placeKey} />
      );
    });

  return (
    <div
      className={`suggestions-list-container${isActive ? ' active' : ''}`}
      aria-hidden={!isActive}
      data-testid={suggestions.container}
    >
      <ul
        className='suggestions-list'
        aria-labelledby='pick-up-locations-suggestions'
        role='listbox'
      >
        {renderSuggestionsResults()}
      </ul>
    </div>
  );
};

export default Suggestions;
