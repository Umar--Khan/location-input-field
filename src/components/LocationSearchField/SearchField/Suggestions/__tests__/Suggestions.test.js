import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import Suggestions from 'components/LocationSearchField/SearchField/Suggestions/index';
import { locationSuggestions } from 'testUtils/mocks';
import { suggestions, tags } from 'testUtils/dataTestIds';
import { NO_RESULTS_FOUND } from 'const';

const suggestionsComponent = (
  <Suggestions suggestionsResults={locationSuggestions} isActive={true} />
);

describe('<Suggestions />', () => {
  it('Render snapshot correctly', () => {
    const component = renderer.create(suggestionsComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Tag label per result', () => {
    const { getAllByTestId } = render(suggestionsComponent);

    const cityTag = getAllByTestId(tags.tagLabel)[0];
    expect(cityTag.textContent).toBe('City');
  });

  it('Amount of results', () => {
    const { getAllByTestId } = render(suggestionsComponent);

    expect(getAllByTestId(suggestions.suggestionItem)).toHaveLength(6);
  });

  it('Correct name and country', () => {
    const { getAllByTestId } = render(suggestionsComponent);

    const suggestionItemName = getAllByTestId(
      suggestions.suggestionItemName,
    )[0];

    const suggestionItemCountry = getAllByTestId(
      suggestions.suggestionItemCountry,
    )[0];

    expect(suggestionItemName.textContent).toBe('KÊ¼oseva');
    expect(suggestionItemCountry.textContent).toBe('Bulgaria');
  });

  it('Render no results from api', () => {
    const noResultApi = [{ name: NO_RESULTS_FOUND, index: 1 }];

    const { getByTestId } = render(
      <Suggestions suggestionsResults={noResultApi} isActive={true} />,
    );

    expect(getByTestId(suggestions.suggestionItem).textContent).toBe(
      NO_RESULTS_FOUND,
    );
  });
});
