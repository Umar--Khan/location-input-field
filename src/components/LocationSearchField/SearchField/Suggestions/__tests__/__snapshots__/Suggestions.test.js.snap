// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Suggestions /> Render snapshot correctly 1`] = `
<div
  aria-hidden={false}
  className="suggestions-list-container active"
  data-testid="suggestions-container"
>
  <ul
    aria-labelledby="pick-up-locations-suggestions"
    className="suggestions-list"
    role="listbox"
  >
    <li
      aria-selected="false"
      className="suggestions-list-item"
      data-testid="suggestion-item"
      id="suggestions-list-item-1"
      role="option"
    >
      <div>
        <span
          className="tag tag--blue"
          data-testid="tags-label"
        >
          City
        </span>
      </div>
      <div
        className="suggestions-list-item__content"
      >
        <p
          className="suggestions-list-item--font-emphasized"
          data-testid="suggestion-item-name"
        >
          Kʼoseva
        </p>
        <p
          className="suggestions-list-item--font-caption"
          data-testid="suggestion-item-country"
        >
          Bulgaria
        </p>
      </div>
    </li>
    <li
      aria-selected="false"
      className="suggestions-list-item"
      data-testid="suggestion-item"
      id="suggestions-list-item-2"
      role="option"
    >
      <div>
        <span
          className="tag tag--blue"
          data-testid="tags-label"
        >
          City
        </span>
      </div>
      <div
        className="suggestions-list-item__content"
      >
        <p
          className="suggestions-list-item--font-emphasized"
          data-testid="suggestion-item-name"
        >
          Kʼochevtsi
        </p>
        <p
          className="suggestions-list-item--font-caption"
          data-testid="suggestion-item-country"
        >
          Bulgaria
        </p>
      </div>
    </li>
    <li
      aria-selected="false"
      className="suggestions-list-item"
      data-testid="suggestion-item"
      id="suggestions-list-item-3"
      role="option"
    >
      <div>
        <span
          className="tag tag--blue"
          data-testid="tags-label"
        >
          City
        </span>
      </div>
      <div
        className="suggestions-list-item__content"
      >
        <p
          className="suggestions-list-item--font-emphasized"
          data-testid="suggestion-item-name"
        >
          Kʼosevo
        </p>
        <p
          className="suggestions-list-item--font-caption"
          data-testid="suggestion-item-country"
        >
          Bulgaria
        </p>
      </div>
    </li>
    <li
      aria-selected="false"
      className="suggestions-list-item"
      data-testid="suggestion-item"
      id="suggestions-list-item-4"
      role="option"
    >
      <div>
        <span
          className="tag tag--blue"
          data-testid="tags-label"
        >
          City
        </span>
      </div>
      <div
        className="suggestions-list-item__content"
      >
        <p
          className="suggestions-list-item--font-emphasized"
          data-testid="suggestion-item-name"
        >
          Kʼosevtsi
        </p>
        <p
          className="suggestions-list-item--font-caption"
          data-testid="suggestion-item-country"
        >
          Bulgaria
        </p>
      </div>
    </li>
    <li
      aria-selected="false"
      className="suggestions-list-item"
      data-testid="suggestion-item"
      id="suggestions-list-item-5"
      role="option"
    >
      <div>
        <span
          className="tag tag--blue"
          data-testid="tags-label"
        >
          City
        </span>
      </div>
      <div
        className="suggestions-list-item__content"
      >
        <p
          className="suggestions-list-item--font-emphasized"
          data-testid="suggestion-item-name"
        >
          Kʼolmen
        </p>
        <p
          className="suggestions-list-item--font-caption"
          data-testid="suggestion-item-country"
        >
          Bulgaria
        </p>
      </div>
    </li>
    <li
      aria-selected="false"
      className="suggestions-list-item"
      data-testid="suggestion-item"
      id="suggestions-list-item-6"
      role="option"
    >
      <div>
        <span
          className="tag tag--blue"
          data-testid="tags-label"
        >
          City
        </span>
      </div>
      <div
        className="suggestions-list-item__content"
      >
        <p
          className="suggestions-list-item--font-emphasized"
          data-testid="suggestion-item-name"
        >
          Kʼochʼara
        </p>
        <p
          className="suggestions-list-item--font-caption"
          data-testid="suggestion-item-country"
        >
          Georgia
        </p>
      </div>
    </li>
  </ul>
</div>
`;
